src/rpm/package.rs

```diff
@@ -220,59 +222,90 @@ impl Package {
-
-        let sig_header_builder = match signer.algorithm() {
-            crate::signature::AlgorithmType::RSA => {
-                sig_header_builder.add_rsa_signature(&header_signature)
-            }
-            crate::signature::AlgorithmType::EdDSA => {
-                sig_header_builder.add_eddsa_signature(&header_signature)
-            }
-            crate::signature::AlgorithmType::ECDSA => {
-                sig_header_builder.add_ecdsa_signature(&header_signature)
-            }
-        };
+        let sig_header = SignatureHeaderBuilder::new()
+            .set_sha256_digest(&header_digest_sha256)
+            .add_openpgp_signature(header_signature)
+            .build()?;
 
-        self.metadata.signature = sig_header_builder.build();
+        self.metadata.signature = sig_header;
         Ok(())
     }
 
     /// Return the key ids (issuers) of the signature as a hexadecimal string
     #[cfg(feature = "signature-pgp")]
     pub fn signature_key_ids(&self) -> Result<Vec<String>, Error> {
-        let mut signature = Err(Error::NoSignatureFound);
-        let rsa_sig = &self
+        let openpgp_sigs = &self
             .metadata
             .signature
-            .get_entry_data_as_binary(IndexSignatureTag::RPMSIGTAG_RSA);
-        if let Ok(rsa_sig) = rsa_sig {
-            signature = Verifier::parse_signature(rsa_sig);
-        }
+            .get_entry_data_as_string_array(IndexSignatureTag::RPMSIGTAG_OPENPGP);
 
-        let eddsa_sig = &self
-            .metadata
-            .signature
-            .get_entry_data_as_binary(IndexSignatureTag::RPMSIGTAG_DSA);
-        if let Ok(eddsa_sig) = eddsa_sig {
-            signature = Verifier::parse_signature(eddsa_sig);
-        }
+        // If RPMSIGTAG_OPENPGP exists, then the other tags (which should contain the same info) are not checked
+        if let Ok(openpgp_sigs) = openpgp_sigs {
+            let mut key_ids: Vec<String> = Vec::new();
 
-        let rpm_v3_sig = &self
-            .metadata
-            .signature
-            .get_entry_data_as_binary(IndexSignatureTag::RPMSIGTAG_PGP);
-        if let Ok(rpm_v3_sig) = rpm_v3_sig {
-            signature = Verifier::parse_signature(rpm_v3_sig);
-        }
+            for base64_sig in openpgp_sigs.iter() {
+                let mut signature = Vec::new();
+                let mut decoder = Base64Decoder::new(Base64Reader::new(base64_sig.as_bytes()));
+                decoder.read_to_end(&mut signature)?;
+                let signature = Verifier::parse_signature(&signature)?;
+
+                let new_key_ids: Vec<String> = signature
+                    .issuer()
+                    .iter()
+                    .map(|x| format!("{:x}", x))
+                    .collect();
+
+                if key_ids.len() != 1 {
+                    return Err(Error::UnexpectedIssuerCount(
+                        key_ids.len().try_into().unwrap(),
+                    ));
```

Comment:
If there's more than one issuer packet, that's a clear error IMO.

If there are zero issuer packets, I'm OK with treating that as an error until I see a sufficient practical reason not to do so.

Reply:
Not really, no